City-Bureau/city-scrapers-philly
# 🏗️ Build spider: Philadelphia Board of Education

## What's this PR do?
Adds a spider to scrape meetings from a Google Calendar managed by the Philadelphia Board of Education (called `phipa_boe`). Also adds a new environment variable to dev and production environments `GOOGLE_CLOUD_API_KEY` in order to use the Google Calendar API. 

## Why are we doing this?
This spider was requested by City Bureau's Philadelphia partners.

## Steps to manually test

#### Set up

In this repo, the `GOOGLE_CLOUD_API_KEY` has already been added to the Github secrets manager so that this spider will work in our `cron` workflow. To successfully execute the spider locally, however, you will need to do the following:

1) Get the `GOOGLE_CLOUD_API_KEY` from the City Bureau secrets manager or a City Bureau technology coordinator.
2) Create a `.env.development` file in the project root:
```
cp .env.development.example .env.development
```
3) Add the key to the file. Replace "<API-VALUE>" with the API key.
```
GOOGLE_CLOUD_API_KEY=<API-VALUE>
```
#### Test
1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spider:
```
scrapy crawl phipa_boe -O test_output.csv
```
4. Monitor the output and ensure no errors are raised.
5. Inspect `test_output.csv` to ensure the data looks valid. Target website is [here](https://spokanevalley.granicus.com/ViewPublisher.php?view_id=3).

## Are there any smells or added technical debt to note? 
There are a lot of helpful meeting materials (agenda, minutes) hosted on a separate [website](https://philasd.novusagenda.com/agendapublic/meetingsgeneral.aspx#) managed by the Philadelphia Board of Education. The page lists meetings by date and meeting title:

![image](https://github.com/City-Bureau/city-scrapers-philly/assets/37225902/393a79cb-eab0-4a6b-b242-5c7c6f1ba37c)

By default, the `phipa_boe` spider returns a generic link to this page for every meeting item. I considered scraping the page itself and merging the materials with the final meeting items but, unfortunately, meeting titles on this page rarely match the titles in the Google Calendar. It would be possible to merge based on meeting date but this seemed like an imperfect approach given there may be occasions where there is more than one meeting in the Google Calendar on a given day. For now, it seemed safer to simply provide a generic link to the page. We could provide more refined handling of meeting materials in a subsequent PR if it's requested by our Philadelphia partners.
